docs/adr/007_cups_firmware_upgrade.md

```diff
@@ -0,0 +1,64 @@
+# 007. CUPS Protocol Implementation - Firmware Upgrade
+
+**Feature**:
+[#1189](https://github.com/Azure/iotedge-lorawan-starterkit/issues/1189)  
+
+**Authors**: Daniele Antonio Maggio, Maggie Salak
+
+**Status**: Proposed
+
+## Overview
+
+Firmware upgrades for LoRa Basics™ Station need to be supported in the CUPS
+protocol implementation, since certain parts of the information exchanged
+between the Basics Station and the CUPS server are tied to a number of
+fields exchanged between the Station and the CUPS server must correspond to the
+current version of the Station.
+
+More information on the protocol and can be found in 
+[The CUPS protocol - documentation][cupsproto].
+
+## In-scope
+
+This document focuses on:
+
+- Defining a flow for executing firmware upgrades of the Basics Station
+- Defining the changes needed in LoRaWan Network Server (LNS) for handling firmware
+  upgrades
+- Defining the changes needed in device twins for the concentrator station
+  stored in IoT Hub for handling firmware upgrades
+- Defining the changes required in the storage solution used for the CUPS
+  protocol implementation
+- Defining the changes needed in LoRa Device Provisioning CLI for allowing
+  firmware upgrades
+
+## Out-of-scope
+
+## Decisions
+
+### Context
+
+The CUPS request described in [The CUPS protocol - documentation][cupsproto]
+contains a number of fields which are related to the current firmware version of
+the Basics Station. Specifically, `station`, `model`, `package` and `keys`
```

Comment:
I think we can keep it simple and only use the package as an indication for the update requirement. If there is a change in station name, we can indicate that with an update of the package version (a new package, will have to be created anyway to deploy a new station, hence, that's somewhat redundant).